ui/StatusQ/src/StatusQ/Components/StatusDateInput.qml

```diff
@@ -0,0 +1,263 @@
+        This property exposes the monthInput for customisation
+    */
+    property alias monthInput: editMonth
+    /*!
+        \qmlproperty alias StatusDateInput::yearInput
+        This property exposes the yearInput for customisation
+    */
+    property alias yearInput: editYear
+
+    /*!
+        \qmlproperty string StatusDateInput::nowText
+        This property holds the now text to be shown on the eidget
+    */
+    property string nowText
+    /*!
+        \qmlproperty double StatusDateInput::presetTimestamp
+        This property holds the timestamp shosen before entering the popup
+    */
+    property double presetTimestamp: Date.now()
+    /*!
+        \qmlproperty bool StatusDateInput::isEditMode
+        This property can turn and off the edit mode for the input
+    */
+    property bool isEditMode: false
+    /*!
+        \qmlproperty bool StatusDateInput::showBackground
+        This property helps turning of the background of the input
+    */
+    property bool showBackground: true
+    /*!
+        \qmlproperty var StatusDateInput::newDate
+        Represents the newly set date in the input
+    */
+    property var newDate
+    /*!
+        \qmlproperty string StatusDateInput::errorMessage
+        This property to assign errorMessage for the date input
+    */
+    property string errorMessage
+    /*!
+        \qmlproperty bool StatusDateInput::isEditMode
+        This property exposes if the input has valid date
+    */
+    readonly property bool valid: editDate.acceptableInput && editMonth.acceptableInput && editYear.acceptableInput
+    /*!
+        \qmlproperty bool StatusDateInput::hasChange
+        This property exposes if the input has been modified by user
+    */
+    readonly property bool hasChange: d.presetDate.valueOf() !== newDate.valueOf()
+    /*!
+        \qmlproperty bool StatusDateInput::hasChange
+        This property exposes if the input has been modified by user
+    */
+    property int supportedStartYear: 0
+
+    /*!
+        \qmlmethod
+        This function resets the input's text
+    */
+    function reset() {
+        d.presetDate = d.getDateWithoutTime(presetTimestamp)
+    }
+    /*!
+        \qmlmethod
+        This function sets the active focus to edit date
+    */
+    function forceActiveFocus() {
+        editDate.forceActiveFocus()
+        editDate.cursorPosition = 0
+    }
+
+    QtObject {
+        id: d
+        readonly property string separator: "/"
+        readonly property bool hasActiveFocus:  editDate.activeFocus || editMonth.activeFocus || editYear.activeFocus
+        readonly property bool isCurrentTimestamp: getDateWithoutTime(Date.now().valueOf()).valueOf() === newDate.valueOf()
+        property var presetDate: d.getDateWithoutTime(presetTimestamp)
+        readonly property bool showError: (!editDate.acceptableInput && !!editDate.text) || (!editMonth.acceptableInput && !!editMonth.text) || !editYear.acceptableInput && !!editYear.text
```

Comment:
```suggestion
        readonly property bool showError: !!editYear.text &&  (!editDate.acceptableInput || !editMonth.acceptableInput || !editYear.acceptableInput)
```

Reply:
not sure what you mean @Cuteivist ?

Reply:
@Khushboo-dev-cpp Sorry for confusion. I did miss that it's for every input.

Anyway there is missing parenthesis